### PR TITLE
Port customizable keybinds from /tg/

### PR DESCRIPTION
**Pull request by ghost**
:cl: Couls and ninjanomnom 
add: You can now customize hotkeys.
add: Also you can bind emotes to keys.
/:cl:

### CODE REVIEW

**Comment by karmaisblackandbluepilled** [comment]
+1
hey emily did something good for once

**Comment by ghost** [comment]
emily has nothing to do with this

**Comment by Rohesie** [comment]
Not everything on this PR is relevant to you, but I'd take the conversion of keybinds to signals if I were you: <https://github.com/tgstation/TerraGov-Marine-Corps/pull/3327>
/tg/ is planning to do it, but hasn't done it yet.
Let me know if you need help with it.

**Comment by ghost** [comment]
@Rohesie stupid question, but what's the advantage of keybinds being signalers? cleaner code, ik.

**Comment by diegoxt45** [comment]
didn't this break tg hard bro

**Comment by ghost** [comment]
@diegoxt45 it broke non-hotkey mode, but that got fixed

i tested this (more than lunaman tests his stuff at least)

**Comment by Rohesie** [comment]
Yea, /tg/ ported an old and incomplete version of our keybinds so it broke a lot of things. Then on the latest referenced PR Coul fixed it. I advised them some.

The advantage of keybinds being signals is in part cleaner code, but it's also that it offers you flexibility. You can register and unregister signals at runtime and you can code things events-based instead of having to do rely on continuous checks or processing. That makes things both better performant and more responsive.

Also the components (or anything else) can listen to the signals, and then the keybind doesn't need to care about managing who's reacting to it or not. The signals will automatically unregister if the listener is deleted, and whatnot.

TL;DR, you _can_ do all this without signals, but it will just be much more work, messier code, and more need to manage things to avoid errors.

**Review by yoyobatty — APPROVED** [review]

**Comment by yoyobatty** [comment]
Okay, if this breaks anything major, I'll break something on you that's major.
